### ISSUE TITLE
README.md should be improved

### ISSUE THREAD

**Issue by zrzka** [resin-io-modules/rendition#609]
### npm ci

Would be nice to have a warning about minimal npm version. I had outdated `npm` version and it fails to run `npm ci`.

### npm run storybook

Check `package.json`, there's no `storybook` script. Should be `npm run start`.

## Components

Almost all tables are _crippled_. One example:

<img width="928" alt="screen shot 2018-09-24 at 08 08 23" src="https://user-images.githubusercontent.com/1084172/45939205-10756800-bfd1-11e8-92c0-55618641a3e2.png">

Many other tables are _crippled_ in the same way.

## Structure

Sections like `Developing`, `Testing` and `Upgrading` should be near the top. Or at least before the `Components` sections. As one has to scroll way too down to get the info how to run it.

**Comment by LucianBuzzo** [comment]
@zrzka That's very strange formatting on the table, since it works with most other rendering tools. I guess no one has noticed it because until recently, the property tables were only exposed through the live storybook and weren't shown in the README.

I agree that we should be specifying supported node/npm versions (I believe npm >=5.7.0 is required for `npm ci`.

With regards to the structure, I disagree with you. For most users, the documentation is more important than development and testing information.

**Comment by LucianBuzzo** [comment]
It appears that `marked` has a more relaxed table syntax than GitHub.
In marked the following is a valid table:

```
| First Header  | Second Header |
| ----------------------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |
```

Whereas GitHub requires `|` symbols in the line below the table header, like this:

```
| First Header  | Second Header |
| ------------- | ------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |
```

**Comment by zrzka** [comment]
> With regards to the structure, I disagree with you. For most users, the documentation is more important than development and testing information.

I agree that most users find docs more important than the development stuff. That's perfectly fine. Maybe we can fix this by just adding TOC at the top? Even manual one?

Something like:

```
# Table of Contents

* [Installation](#installation)
* [Usage](#usage)
* [Components](#components)
* [Development](#developing)
* [Testing](#testing)
```

Or move the dev & testing into a separate doc and link it at the top?

**Comment by LucianBuzzo** [comment]
@zrzka The table of contents sounds like a good idea, I'd consider moving the dev and testing into a seperate doc if its starts getting weightier.
